ops/model.py

```diff
@@ -559,8 +559,10 @@ def __init__(self, network_info: dict):
         # interfaces with the same name.
         for interface_info in network_info.get('bind-addresses', []):
             interface_name = interface_info.get('interface-name')
-            for address_info in interface_info.get('addresses', []):
-                self.interfaces.append(NetworkInterface(interface_name, address_info))
+            addrs = interface_info.get('addresses')
+            if addrs is not None:
+                for address_info in addrs:
```

Comment:
To avoid the extra `if` and nesting, I quite like the pattern `for address_info in addrs or []:`